small/small.c

```diff
@@ -281,19 +281,14 @@ small_mempool_create(struct small_alloc *alloc)
 						    pool - 1);
 			cur_order_pool = pool;
 		}
-		/*
-		 * Maximum object size for mempool allocation ==
-		 * alloc->objsize_max. If we have reached this size,
-		 * there will be no more pools - loop will be broken
-		 * at the next iteration. So we need to create the last
-		 * group of pools.
-		 */
-		if (objsize == alloc->objsize_max) {
-			assert(cur_order_pool->pool.slab_ptr_mask ==
-			       pool->pool.slab_ptr_mask);
-			small_mempool_create_groups(alloc, cur_order_pool,
-						    pool);
-		}
+	}
+	if (alloc->small_mempool_cache_size > 0) {
```

Comment:
This check seems useless - we can't possibly stop without running at least one iteration.

Reply:
Fixed.